lib/azure/armrest/model/hash_model.rb

```diff
@@ -0,0 +1,140 @@
+require 'active_support/core_ext/string/inflections'
+
+module Azure
+  module Armrest
+    class HashModel
+      class << self
+        private
+
+        # Convert a key to a model class attribute which must be a qualified ruby method name
+        # key is from the original JSON or Hash object
+        #
+        # This is the default implementation, to be overridden by subclasses.
+        #
+        def key_to_attribute(key)
+          key.to_s.underscore
+        end
+
+        # Should setter methods be automatically created?
+        #
+        # Default to true, to be overridden by subclasses.
+        #
+        def generate_writer?
+          true
+        end
+
+        # Should values of this attribute be excluded from converting to model objects
+        #
+        # Default to false, to be overridden by subclasses.
+        #
+        def attr_excluded?(_attr_name)
+          false
+        end
+
+        # Declare attributes that need to be converted to Model
+        def attr_model(*attrs)
+          # Merge the declared attributes to the modelized list
+          @modelized_list = modelized_list | Set.new(attrs.map(&:to_s))
+        end
+
+        # Return the list of modelized attributes
+        def modelized_list
+          @modelized_list ||= Set.new
+        end
+
+        def convert_value(key, value, parent)
```

Comment:
So, this is still going to be costly to run, since there is a lot of `#underscore`, `#camelize`, and `#singularize` that will have to happen for each model instantiated, which was a lot of what I was trying to avoid in https://github.com/ManageIQ/azure-armrest/pull/319

Not sure of a good way around this though, without doing the approach in #319, and I fully admit that PR is kinda ugly and has it's own flaws as well.

Reply:
I think the way around it is to instead to implement "as_hash" methods on ResourceGroupBasedService that users can use when they just want a plain hash and don't need intermediate objects.

So, you would have `get_as_hash`, `list_as_hash`, etc.

Reply:
Sure, I like that, or maybe just an option in those methods to avoid the object instantiation portion (since everything else should be the same).

Reply:
And I guess not being familiar with the codebase myself, I didn't realize that the service models are the only things involved with making the requests to the API, where I was under the "Rails" mind set that the models would also be in charge of making calls.  That means doing what you are suggesting here really wouldn't be that involved.  Cool!

Reply:
To make things interesting, the StorageAccount model *does* make requests, but it's the only model that does. That, however, could change in the future.

Reply:
Darn it!  I thought there was a model that did that!

Anyway, looks like that is an exception to the rule, and not the standard.